### PR TITLE
Login access

### PR DESCRIPTION
**Pull request by rickwolox**
### Summary
Creation of service for validation of User when clicking Log In button

### Screenshot
-
 
### Trello Card
https://trello.com/c/3JRFHGtX/19-u2-backend-login-acceso

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/BaseConfiguration.kt [hunk 1]
```diff
@@ -4,6 +4,7 @@ internal open class BaseConfiguration {
 
     companion object {
         const val EXAMPLE_CONFIGURATION_KEY = "https://jsonplaceholder.typicode.com"
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/BaseConfiguration.kt]: If you're not using this key, you can remove it

File: app/src/main/java/ar/com/wolox/android/example/model/User.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ *
+ */
+public class User {
+
+    @SerializedName("id")
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/model/User.java:13]: Oh my eyes :roll_eyes: Try to write attributes on a single line. For example: `@SerializedName("id") @Expose private Integer id;`

File: app/src/main/java/ar/com/wolox/android/example/model/User.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    @Expose
+    private String location;
+    @SerializedName("name")
+    @Expose
+    private String name;
+    @SerializedName("phone")
+    @Expose
+    private String phone;
+
+    public Integer getId() {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/model/User.java:34]: Add nullabilities here and in the rest of methods!

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java [hunk 4]
```diff
@@ -76,8 +87,27 @@ private void restoreFormOnInit() {
         }
     }
 
+    private void validateUser(String username, String password) {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java]: Nullabilities here!

**Review by lmiotti — CHANGES_REQUESTED** [review]
This is very good, nonetheless there are some things to be solved. Keep on like this!

File: app/src/main/java/ar/com/wolox/android/example/model/User.java (hunk 2 above)
Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/example/model/User.java:13]: Hey! this is still the same

**Review by lmiotti — CHANGES_REQUESTED** [review]
There's a correction remaining.

File: app/src/main/java/ar/com/wolox/android/example/model/User.java (hunk 2 above)
Comment (rickwolox) [reply app/src/main/java/ar/com/wolox/android/example/model/User.java:13]: Sorry, bad news, checkstyle didn't let me to simplify in a single line, so I had to take two :(

Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/example/model/User.java:13]: Ok, I'm starting to hate checkstyle :smile:

**Review by lmiotti — APPROVED** [review]
great

**Pull request merged**